Sources/TuistGenerator/Generator/DescriptorGenerator.swift

```diff
@@ -0,0 +1,85 @@
+import Basic
+import Foundation
+import TuistCore
+import TuistSupport
+import XcodeProj
+
+public struct GeneratedSchemeDescriptor {
+    public var scheme: XCScheme
+    public var shared: Bool
+}
+
+public enum GeneratedSideEffect {
+    case file(GeneratedFile)
```

Comment:
Should we have a side effect that ensures that a directory has a given state? For example, in order to create a file under a given path, we must ensure that its parent directory exists:

```swift
GeneratedSideEffect.directory(path, .exists)
```

Notice that there's a second argument to specify what should be the state of the directory. [Ansible](https://docs.ansible.com/ansible/latest/modules/file_module.html#parameter-state) does it like that. You can say there must be a directory at the path X, or the file at the path Y should have this content. What is necessary to take the file or the directory to that state is really up to the utility that resolves the descriptor.

Reply:
Sure, I've added the Ansible like side effects (e.g. `GeneratedFile(.. state: .absent)` / `GeneratedFile(... state: .present)`

Currently the writer will ensure to create directories for the specified files.